sgkit/tests/test_window.py

```diff
@@ -0,0 +1,46 @@
+import allel
+import dask.array as da
+import numpy as np
+import pytest
+
+from sgkit.window import moving_statistic
+
+
+@pytest.mark.parametrize(
+    "length, chunks, size, step", [(12, 6, 4, 4), (12, 6, 4, 2), (12, 5, 4, 4)]
+)
+def test_moving_statistic_1d(length, chunks, size, step):
+    values = da.from_array(np.arange(length), chunks=chunks)
+
+    stat = moving_statistic(values, np.sum, size=size, step=step, dtype=values.dtype)
+    stat = stat.compute()
+    if length % size != 0 or size != step:
+        # scikit-allel misses final window in this case
+        stat = stat[:-1]
+
+    values_sa = np.arange(length)
+    stat_sa = allel.moving_statistic(values_sa, np.sum, size=size, step=step)
+
+    np.testing.assert_equal(stat, stat_sa)
+
```

Comment:
I think a test for dtype preservation when provided float32 inputs would be good for these kinds of functions.

Reply:
I added an assert to check for dtype preservation here.